plumbing/transport/internal/common/common.go

```diff
@@ -46,6 +46,8 @@ type Commander interface {
 type Command interface {
 	// SetAuth sets the authentication method.
 	SetAuth(transport.AuthMethod) error
+	// Perform the actual network connection.
+	Connect() error
```

Comment:
Ah, I had wondered about that. I've moved `AuthMethod` to the `PackSession` functions now.

While doing that, I noticed that server also uses these functions. I've left a `TODO` for now to design / implement `AuthMethod` for servers. I *think* this could be used to verify a client connection to the go-git server in the future.